app/src/main/java/com/example/admin/lab_50_82/MultithreadingActivity.java

```diff
@@ -0,0 +1,216 @@
+                adapter_.notifyDataSetChanged();
+                new addItemViaAsyncTask().execute();
+                onOffButtons(false);
+            }
+        });
+
+        launchThread_.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                list_.clear();
+                list_.add(getResources().getString(R.string.added_label));
+                adapter_.notifyDataSetChanged();
+                addItem();
+                onOffButtons(false);
+            }
+        });
+
+        launchThreadViaHandler_.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                list_.clear();
+                list_.add(getResources().getString(R.string.added_label));
+                adapter_.notifyDataSetChanged();
+                addItemUsingHandler();
+                onOffButtons(false);
+            }
+        });
+    }
+
+    public class addItemViaAsyncTask extends AsyncTask<String, Integer, Void> {
+
+        @Override
+        protected Void doInBackground(String... params) {
+
+            for (int i = 1; i <= 10; i++) {
+                try {
+                    Thread.sleep(1000);
+                    publishProgress(i);
+                } catch (InterruptedException e) {
+                    e.printStackTrace();
+                }
+            }
+            return null;
+        }
+
+        @Override
+        protected void onProgressUpdate(Integer... value) {
+            updateAdapter(value[0]);
+            if (isItTenthItem(value[0])) {
+                onOffButtons(true);
+            }
+        }
+    }
+
+    public void addItem() {
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                for (j_ = 0; j_ < 10; j_++) {
+                    try {
+                        Thread.sleep(1000);
+                    } catch (InterruptedException e) {
+                        Log.e(TAG, e.toString());
+                    }
+
+                    runOnUiThread(new Runnable() {
+                        @Override
+                        public void run() {
+                            updateAdapter(j_);
+                            if (isItTenthItem(j_)) {
+                                onOffButtons(true);
+                            }
+                        }
+                    });
+                }
+            }
+        }).start();
+    }
+
+    public void addItemUsingHandler() {
+        new Thread(new Runnable() {
+            @Override
+            public void run() {
+                for (int i = 1; i <= 10; i++) {
+                    try {
+                        Thread.sleep(1000);
+                        handler_.sendEmptyMessage(i);
+                    } catch (InterruptedException e) {
+                        Log.e(TAG, e.toString());
+                    }
+                }
+            }
+        }).start();
```

Comment:
I used such way   handler_.sendMessageDelayed(handler_.obtainMessage(i),1000);
But in this case after 1000 ms the user will see just 10. Because the message sends with delay.